etest_test/profile_test.py

```diff
@@ -0,0 +1,221 @@
+    for arch in ["armv5", "armv6", "armv7"]:
+        # Alternative libcs
+        for libc in ["musl", "uclibc"]:
+            assert try_invalid_profile(logger, False, arch, libc, False, True, False)
+
+        # Hardened
+        assert try_invalid_profile(logger, False, arch, "glibc", True, True, False)
+
+        # Systemd
+        assert try_invalid_profile(logger, False, arch, "glibc", False, True, True)
+
+
+def test_invalid_arm64_profiles() -> None:
+    """Ensure invalid ARM profiles raise exceptions."""
+    arch = "arm64"
+
+    # Alternative libcs
+    for libc in ["musl", "uclibc"]:
+        assert try_invalid_profile(logger, False, arch, libc, False, True, False)
+
+    # Hardened
+    assert try_invalid_profile(logger, False, arch, "glibc", True, True, False)
+
+
+def test_invalid_x86_profiles() -> None:
+    """Ensure invalid x86 profiles raise exceptions."""
+    arch = "x86"
+
+    # Hardened musl
+    assert try_invalid_profile(logger, False, arch, "musl", True, True, False)
+
+
+def test_invalid_ppc64_profiles() -> None:
+    """Ensure invalid ppc64 profiles raise exceptions."""
+    arch = "ppc64"
+
+    # uclibc
+    assert try_invalid_profile(logger, False, arch, "uclibc", False, True, False)
+
+    # Vanilla musl
+    assert try_invalid_profile(logger, False, arch, "musl", False, True, False)
+
+    # Hardened
+    assert try_invalid_profile(logger, False, arch, "glibc", True, True, False)
+
+
+def test_systemd() -> None:
+    """Ensure systemd profiles work correctly."""
+    for arch in ["amd64", "x86", "arm64", "ppc64"]:
+        # Base
+        result = sut.Profile(logger, False, arch, "glibc", False, True, True)
+        assert result.profile == f"{arch}-systemd"
+
+        # Alternative libcs
+        for libc in ["musl", "uclibc"]:
+            assert try_invalid_profile(logger, False, arch, libc, False, True, True)
+
+        # Hardened
+        assert try_invalid_profile(logger, False, arch, "glibc", True, True, True)
+
+    # No multilib
+    assert try_invalid_profile(logger, False, "amd64", "glibc", False, False, True)
+
+
+def test_no_multilib() -> None:
+    """Ensure no-multilib only affects AMD64."""
+    # Other arches
+    for arch in ["x86", "armv5", "armv6", "armv7", "arm64", "ppc64"]:
+        assert try_invalid_profile(logger, True, arch, "glibc", False, False, False)
+
+    arch = "amd64"
+
+    # Base
+    result = sut.Profile(logger, False, arch, "glibc", False, False, False)
+    assert result.profile == f"{arch}-nomultilib"
+
+    # Hardened
+    result = sut.Profile(logger, False, arch, "glibc", True, False, False)
+    assert result.profile == f"{arch}-hardened-nomultilib"
+
+    # Alternative libcs
+    for libc in ["musl", "uclibc"]:
+        assert try_invalid_profile(logger, False, arch, libc, False, False, False)
+
+    # Systemd
+    assert try_invalid_profile(logger, False, arch, "glibc", False, False, True)
+
+
+def test_warnings() -> None:
+    """Ensure warnings happen correctly."""
```

Comment:
This tests that warnings are correct but then only asserts no warnings.  That seems backwards.  If we have a test of the warnings we should assert those warnings.

Reply:
It asserts that it fails when providing strict on stuff that gives a warning and that it doesnt fail when not providing that. Its a more straight forward way than actually checking for "WARNING" in the output, and I believe it works the same. Let me know if that makes any sense.

Reply:
Perhaps we just need to make the test name and description a bit more clear to make that story make sense.  Let's try updating those and I'll review this again in the next review pass.